### PR TITLE
*: tiny update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Review CONTRIBUTING.md, tiny update.

### What is changed and how it works?

After we migrate to Go module, the Makefile still there and some tools depends on GOPATH.
So we haven't get rid of GOPATH totally.

### Check List 

 - No code

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8814)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,7 +63,25 @@ please [set one up](http://golang.org/doc/code.html).
 
 The version of GO should be **1.11** or above.
 
-After installation, you'll need `GOPATH` defined,
+After installation, there are two ways to build TiDB binary.
+
+#### For a quick taste
```
Comment (shenli) [inline CONTRIBUTING.md]: What is "quick taste"?

Comment (tiancaiamao) [reply CONTRIBUTING.md]: I mean this is for those who just want to know how to compile TiDB binary from the source, not seriously using it for development.
And I guess the word "quick taste" is a perfectly match here, maybe it means "尝鲜" in chinese?
https://www.percona.com/blog/2017/12/05/mysql-8-0-window-functions-a-quick-taste/

**Comment by tiancaiamao** [comment]
PTAL @shenli

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -63,7 +63,25 @@ please [set one up](http://golang.org/doc/code.html).
 
 The version of GO should be **1.11** or above.
 
-After installation, you'll need `GOPATH` defined,
+After installation, there are two ways to build TiDB binary.
+
+#### A quick taste
```
Comment (shenli) [inline CONTRIBUTING.md]: For a quick taste

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -63,7 +63,25 @@ please [set one up](http://golang.org/doc/code.html).
...
+echo 'module tidb' > go.mod
+GO111MODULE=on go get github.com/pingcap/tidb@c385cbdcca83eeed283139814a7ea149e3116e66
+GO111MODULE=on go build -o tidb-server github.com/pingcap/tidb/tidb-server
+```
+
+The `c385cbdcca83eeed283139814a7ea149e3116e66` could be changed to the other commit hash, try the latest commit hash [here](https://github.com/pingcap/tidb/commits/master).
+
+#### For development
+
+You'll need `GOPATH` defined,
```
Comment (shenli) [inline CONTRIBUTING.md]: Merge line 85 with line 84.

**Comment by shenli** [comment]
@lilin90 PTAL

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -63,7 +63,25 @@ please [set one up](http://golang.org/doc/code.html).
...
+
+```
+mkdir tmp
+cd tmp
+echo 'module tidb' > go.mod
+GO111MODULE=on go get github.com/pingcap/tidb@c385cbdcca83eeed283139814a7ea149e3116e66
+GO111MODULE=on go build -o tidb-server github.com/pingcap/tidb/tidb-server
+```
+
+The `c385cbdcca83eeed283139814a7ea149e3116e66` could be changed to the other commit hash, try the latest commit hash [here](https://github.com/pingcap/tidb/commits/master).
```
[the remark this answers is not in the document]
Comment (tiancaiamao) [reply CONTRIBUTING.md]: Done

**Comment by tiancaiamao** [comment]
PTAL @shenli @lilin90

File: CONTRIBUTING.md (hunk 1 above)
Comment (morgo) [reply CONTRIBUTING.md]: I would say, _a quick spin_, but a quick taste works for me too.

**Comment by tiancaiamao** [comment]
PTAL @shenli

**Comment by tiancaiamao** [comment]
PTAL @shenli

**Comment by tiancaiamao** [comment]
PTAL @shenli

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -63,8 +63,25 @@ please [set one up](http://golang.org/doc/code.html).
...
-After installation, you'll need `GOPATH` defined,
-and `PATH` modified to access your Go binaries.
+After installation, there are two ways to build TiDB binary.
+
+#### For a quick taste
+
+The `GOPATH` is not necessary.
+
+```
+mkdir tmp
```
Comment (zz-jason) [inline CONTRIBUTING.md]: why not using `git clone`?

Comment (zz-jason) [reply CONTRIBUTING.md]:
```sh
git clone https://github.com/pingcap/tidb.git
cd tidb
make
```

Comment (tiancaiamao) [reply CONTRIBUTING.md]: Makefile is convenient, but if you don't want it, `go get` is a good substitution, especially for gophers!

Comment (zz-jason) [reply CONTRIBUTING.md]: Yeah. How about adding another section for non gophers about how to have a quick taste? Maybe they will fall in love with golang from this first beginning?

Comment (tiancaiamao) [reply CONTRIBUTING.md]: @zz-jason 
Those steps are for non gophers:
https://github.com/pingcap/tidb/pull/8814/files#diff-6a3371457528722a734f3c51d9238c13R97

Comment (tiancaiamao) [reply CONTRIBUTING.md]: 1. It doesn't need clone the repo
2. It doesn't require GOPATH

That's a big deal comparing to `git clone+make`
You know, the single `make` command still need to get go module  dependency, so it's much heavier.
@zz-jason

Comment (zz-jason) [reply CONTRIBUTING.md]: Okay

**Pull request merged**